src/flows/shared/steps/OrganizeCollection/Editor/CollectionEditor.tsx

```diff
@@ -30,8 +31,22 @@ function convertNftsToEditModeNfts(nfts: Nft[], isSelected = false): EditModeNft
   }));
 }
 
+function insertWhitespaceBlocks(
+  stagingItems: StagingItem[],
+  whitespaceList: number[]
+): StagingItem[] {
+  const result = [...stagingItems];
+  // Insert whitespace blocks into the list of items to stage according to the saved whitespace indexes.
+  // Offset the index to insert at by the number of whitespaces already added
+  whitespaceList.forEach((index, offset) =>
+    result.splice(index + offset, 0, { id: `blank-${generate12DigitId()}` })
+  );
+
+  return result;
+}
```

Comment:
should we abstract this? logic used elsewhere

Reply:
done